### PR TITLE
Fix unaligned memory access on LNS

### PR DESCRIPTION
**Pull request by fbeutin-ledger**

### CODE REVIEW

**Review by xchapron-ledger — COMMENTED** [review]
I think we should get rid on this packed trick, but this can be a fix meanwhile.

**Review by xchapron-ledger — DISMISSED** [review]

File: src/set_partner_key.c [hunk 1]
```diff
@@ -73,10 +73,16 @@ int set_partner_key(const command_t *cmd) {
...
+    memset(G_swap_ctx.partner.prefixed_name, 0, sizeof(G_swap_ctx.partner.prefixed_name));
+    // Prepare the prefix for FUND display
+    memcpy(&G_swap_ctx.partner.prefixed_name,
+           PARTNER_NAME_PREFIX_FOR_FUND,
+           sizeof(PARTNER_NAME_PREFIX_FOR_FUND));
+    // The incoming partner name is NOT NULL terminated, so we use memcpy.
+    // Don't erase the prefix copied above
+    memcpy(G_swap_ctx.partner.prefixed_name + sizeof(PARTNER_NAME_PREFIX_FOR_FUND) - 1,
+           partner.bytes,
+           partner.size);
```
Comment (xchapron-ledger) [inline src/set_partner_key.c:85]: I think you could safely use `strlcpy` and `strlcat` or maybe `snprintf` to make the construct more obvious but this is equivalent.

File: src/globals.h [hunk 2]
```diff
@@ -37,23 +37,12 @@ extern uint8_t G_io_seproxyhal_spi_buffer[IO_SEPROXYHAL_BUFFER_SIZE_B];
 #define MAX_PARTNER_NAME_LENGHT 15
 
 #define PARTNER_NAME_PREFIX_FOR_FUND "To "
-#define PARTNER_NAME_PREFIX_SIZE     (sizeof(PARTNER_NAME_PREFIX_FOR_FUND) - 1)  // Remove trailing '\0'
 
-#pragma pack(push, 1)
 typedef struct partner_data_s {
     uint8_t name_length;
```
Comment (xchapron-ledger) [inline src/globals.h:42]: This field doesn't seem to be used anymore

File: src/set_partner_key.c [hunk 3]
```diff
@@ -73,11 +73,14 @@ int set_partner_key(const command_t *cmd) {
...
-    memcpy(G_swap_ctx.partner.name, partner.bytes, partner.size);
-    G_swap_ctx.partner.name_length = partner.size;
+    memset(G_swap_ctx.partner.prefixed_name, 0, sizeof(G_swap_ctx.partner.prefixed_name));
+    // Prepare the prefix for FUND display
+    strlcpy(G_swap_ctx.partner.prefixed_name,
+            PARTNER_NAME_PREFIX_FOR_FUND,
+            sizeof(G_swap_ctx.partner.prefixed_name));
+    // The incoming partner name is NOT NULL terminated, so we use strncat
+    // Don't erase the prefix copied above
+    strncat(G_swap_ctx.partner.prefixed_name, (char *) partner.bytes, partner.size);
```
Comment (xchapron-ledger) [inline src/set_partner_key.c:83]: Do we have a check somewhere that `partner.size <= MAX_PARTNER_NAME_LENGHT`?

**Review by xchapron-ledger — APPROVED** [review]

File: src/set_partner_key.c (hunk 3 above)
Comment (fbeutin-ledger) [reply src/set_partner_key.c:83]: Yes, the parser also checks boundaries

**Pull request merged**